papermill/tests/test_execute.py

```diff
@@ -45,6 +45,23 @@ def test_quoted_params(self):
         self.assertEqual(test_nb.node.cells[1].get('source'), u'# Parameters\nmsg = "\\"Hello\\""\n')
         self.assertEqual(test_nb.parameters, {'msg': '"Hello"'})
 
+    def test_backslash_params(self):
+        execute_notebook(self.notebook_path, self.nb_test_executed_fname, {'foo': r'do\ not\ crash'})
+        test_nb = read_notebook(self.nb_test_executed_fname)
+        self.assertEqual(test_nb.node.cells[1].get('source'), u'# Parameters\nfoo = "do\\\\ not\\\\ crash"\n')
+        self.assertEqual(test_nb.parameters, {'foo': 'do\\ not\\ crash'})
+
+    def test_backslash_quote_params(self):
+        execute_notebook(self.notebook_path, self.nb_test_executed_fname, {'foo': r'bar=\"baz\"'})
+        test_nb = read_notebook(self.nb_test_executed_fname)
+        self.assertEqual(test_nb.node.cells[1].get('source'), u'# Parameters\nfoo = "bar=\\\\\\"baz\\\\\\""\n')
+        self.assertEqual(test_nb.parameters, {'foo': 'bar=\\\"baz\\\"'})
+
+    def test_double_backslash_quote_params(self):
+        execute_notebook(self.notebook_path, self.nb_test_executed_fname, {'foo': r'\\"bar\\"'})
+        test_nb = read_notebook(self.nb_test_executed_fname)
+        self.assertEqual(test_nb.node.cells[1].get('source'), u'# Parameters\nfoo = "\\\\\\\\\\"bar\\\\\\\\\\""\n')
+        self.assertEqual(test_nb.parameters, {'foo': '\\\\\"bar\\\\\"'})
```

Comment:
These tests might be a little more readable with a raw string (or do you have to double escape there too)?

Reply:
It's one less level of escaping, but the parameters with `\n` can't use raw strings and match correctly. Though I could split on `\n` and compare lists.